RecoParticleFlow/PFProducer/plugins/kdtrees/KDTreeLinkerTrackHcal.cc

```diff
@@ -122,43 +122,42 @@ KDTreeLinkerTrackHcal::searchLinks()
     // The track didn't reach hcal
     if( ! atHCAL.isValid()) continue;
     
-    double dHEta = atHCALExit.positionREP().Eta() - atHCAL.positionREP().Eta();
-    double dHPhi = atHCALExit.positionREP().Phi() - atHCAL.positionREP().Phi(); 
-    if ( dHPhi > M_PI ) dHPhi = dHPhi - 2. * M_PI;
-    else if ( dHPhi < -M_PI ) dHPhi = dHPhi + 2. * M_PI; 
+    double dHeta = atHCALExit.positionREP().eta() - atHCAL.positionREP().eta();
+    double dHphi = atHCALExit.positionREP().phi() - atHCAL.positionREP().phi(); 
+    if ( dHphi > M_PI ) dHphi = dHphi - 2. * M_PI;
+    else if ( dHphi < -M_PI ) dHphi = dHphi + 2. * M_PI; 
     
-    double tracketa = atHCAL.positionREP().Eta() + 0.1 * dHEta;
-    double trackphi = atHCAL.positionREP().Phi() + 0.1 * dHPhi;
+    double tracketa = atHCAL.positionREP().eta() + 0.1 * dHeta;
+    double trackphi = atHCAL.positionREP().phi() + 0.1 * dHphi;
     
     if (trackphi > M_PI) trackphi -= 2 * M_PI;
     else if (trackphi < -M_PI) trackphi += 2 * M_PI;
 
     // Estimate the maximal envelope in phi/eta that will be used to find rechit candidates.
     // Same envelope for cap et barrel rechits.
     double inflation = 1.;
-    double rangeEta = (getCristalPhiEtaMaxSize() * (1.5 + 0.5) + 0.2 * fabs(dHEta)) * inflation; 
-    double rangePhi = (getCristalPhiEtaMaxSize() * (1.5 + 0.5) + 0.2 * fabs(dHPhi)) * inflation; 
+    double rangeeta = (getCristalPhiEtaMaxSize() * (1.5 + 0.5) + 0.2 * fabs(dHeta)) * inflation; 
+    double rangephi = (getCristalPhiEtaMaxSize() * (1.5 + 0.5) + 0.2 * fabs(dHphi)) * inflation; 
 
     // We search for all candidate recHits, ie all recHits contained in the maximal size envelope.
     std::vector<KDTreeNodeInfo> recHits;
-    KDTreeBox trackBox(tracketa - rangeEta, tracketa + rangeEta, 
-		       trackphi - rangePhi, trackphi + rangePhi);
+    KDTreeBox trackBox(tracketa - rangeeta, tracketa + rangeeta, 
+		       trackphi - rangephi, trackphi + rangephi);
     tree_.search(trackBox, recHits);
     
     // Here we check all rechit candidates using the non-approximated method.
     for(std::vector<KDTreeNodeInfo>::const_iterator rhit = recHits.begin(); 
 	rhit != recHits.end(); ++rhit) {
 
-      const reco::PFRecHit::REPPoint &rhrep		   = rhit->ptr->positionREP();
-      const std::vector<reco::PFRecHit::REPPoint>& corners = rhit->ptr->getCornersREP();
-      if(corners.size() != 4) continue;
```

Comment:
corners are always size 8. PF uses only the front 4.
look at CaloCellGeometry, line 69